### PR TITLE
[Compute] `az sig image-version create/update`: Add `--allow-replicated-location-deletion` to support removing gallery image version from replicated regions

### PR DESCRIPTION
**Pull request by PARADISSEEKR**
**Related command**

`az sig image-version create`
`az sig image-version update`

**Description**

`az sig image-version --allow-replicated-location-deletion`
close https://github.com/Azure/azure-cli/issues/24099

**Testing Guide**

**History Notes**

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/vm/custom.py [hunk 1]
```diff
@@ -4642,6 +4642,10 @@ def create_image_version(cmd, resource_group_name, gallery_name, gallery_image_n
                                                             data_disk_images=data_disk_images)
         image_version = ImageVersion(publishing_profile=profile, location=location, tags=(tags or {}),
                                      storage_profile=storage_profile)
+        if cmd.supported_api_version(min_api='2022-03-03', operation_group='gallery_image_versions'):
+            GalleryImageVersionSafetyProfile = cmd.get_models('GalleryImageVersionSafetyProfile')
+            image_version.safety_profile = GalleryImageVersionSafetyProfile(
+                allow_deletion_of_replicated_locations=allow_replicated_location_deletion)
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/custom.py:4648]: Could we put the min api-version control in the parameter registration?

File: src/azure-cli/azure/cli/command_modules/vm/custom.py [hunk 2]
```diff
@@ -4688,6 +4693,9 @@ def update_image_version(cmd, resource_group_name, gallery_name, gallery_image_n
         image_version.publishing_profile.replica_count = replica_count
     if image_version.storage_profile.source is not None:
         image_version.storage_profile.os_disk_image = image_version.storage_profile.data_disk_images = None
+    if cmd.supported_api_version(min_api='2022-03-03', operation_group='gallery_image_versions'):
+        if image_version.safety_profile.allow_deletion_of_replicated_locations:
+            image_version.safety_profile.allow_deletion_of_replicated_locations = allow_replicated_location_deletion
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/custom.py:4698]: Same as above

**Comment by yonzhan** [comment]
Compute

**Review by zhoxing-ms — APPROVED** [review]

**Comment by msandu62** [comment]
Is this available in the sdk for python also? or packer during the image build? cannot find the option (in packer at least)

**Pull request merged**